Functions/xASL_io_ReadDataPar.m

```diff
@@ -0,0 +1,250 @@
+function [x] = xASL_io_ReadDataPar(DataParFile)
+% xASL_io_ReadDataPar This function reads DATA_PAR .json or .m file and creates the x structure.
+%
+% FORMAT:   [x] = xASL_io_ReadDataPar(DataParFile)
+%
+% INPUT:
+%   DataParFile     - Filename of the DATA_PAR file with either .m or .json extension
+%
+% OUTPUT:
+%   x               - x structure
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION:  This function reads in a DATA_PAR .json or .m file and creates the x
+%               structure. The name of the DATA_PAR file is given as a string or
+%               character array. The output is the x structure. It only loads the data, removes the x-prefixes, 
+%               but keeps all the field names and units. It doesn't do any conversions to or from BIDS. The 
+%               only added value to normal json-read is that it detects invalid entries (numbers in strings, and 
+%               weird arrays), converts them correctly and reports this incorrect entries so that they can be manually 
+%               fixed. Also, if an .m file is provided, it converts and saves it to a JSON file (doesn't overwrite)
+%               and reports that you should stop using .m files.
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+%
+% EXAMPLE:      xASL_io_ReadDataPar('DataParFile.json')
+%
+% EXAMPLE 2:    xASL_io_ReadDataPar('DataParFile.m')
+%
+% EXAMPLE 2:    JSON FILE
+%
+% {
+% 	"x": [{
+% 			"name": 				"ExampleDataSet",
+% 			"subject_regexp": 		"^Sub-\\d{3}$",
+% 			"M0": 					"separate_scan",
+% 			"BackgroundSuppressionNumberPulses": 2,
+% 			"readout_dim": 			"2D",
+% 			"QUALITY": 				false,
+% 			"Vendor": 				"Philips",
+% 		}]
+% }
+%
+% Don't forget to escape the backslashes!
```

Comment:
Can we add an automatic escaping of backslashes (including a warning as well), at least for the `subject_regexp` field, which is generally the problem? Saves us some user questions ;)

Reply:
I wouldn't add automatic escaping. Because you will never know if the backslash should be escaped, or if it was meant to escape some other character ;)

Reply:
I still think the automatic escaping could be very nice, only for this field, and only when used in combination with the digits (then you know if it should or shouldn't be escaped).